src/app/core/services/geojson/geojson.service.ts

```diff
@@ -47,9 +47,30 @@ export class GeoJSONService {
     return this.db.get<GeoJSONItem[]>('geojson-metadata');
   }
 
+  /**
+   * Updates metadata for a given item
+   * @param item the geojson item to update
+   */
+  async updateMetadata(item: GeoJSONItem) {
+    try {
+      const items = this.metadata();
+      const index = items.findIndex((i) => i.id === item.id);
+      if (index === -1) {
+        throw new Error(`Item with id ${item.id} not found`);
+      }
+
+      const updatedItems = items.map((m) => (m.id === item.id ? { ...m, ...item } : m));
+      await this.saveMetadata(updatedItems);
+      this.metadata.set(updatedItems);
+    } catch (error) {
+      this.logger.error(error, DEBUG_TAG, 'Could not update metadata', { item });
+      throw error;
+    }
+  }
```

Comment:
```suggestion
  updateMetadata(item: GeoJSONItem) {
    this.metadata.update((items) => {
      const other = items.filter((x) => x.id !== item.id);
      return [...other, item];
    });
  }
```

Kan hende det er et problem å ikke sjekke for om den eksisterer, men i hvilke tilfeller vil det kunne feile? I så fall kan vi legge til det, men denne trenger uansett ikke være async.